cmake-proxies/cmake-modules/AudacityTesting.cmake

```diff
@@ -0,0 +1,143 @@
+if( ${_OPT}has_tests )
+   if( CMAKE_CONFIGURATION_TYPES )
+      set( TESTS_DIR "${CMAKE_BINARY_DIR}/tests" )
+      set( TESTS_DEST_DIR "${CMAKE_BINARY_DIR}/tests/${CMAKE_CFG_INTDIR}" )
+   else()
+      set( TESTS_DIR "${CMAKE_BINARY_DIR}/tests" )
+      set( TESTS_DEST_DIR "${CMAKE_BINARY_DIR}/tests/${CMAKE_BUILD_TYPE}" )
+   endif()
+
+   # Install Catch2 framework
+
+   add_conan_lib(
+      Catch2
+      catch2/2.13.8
+      REQUIRED
+      ALWAYS_ALLOW_CONAN_FALLBACK
+   )
+
+   # Setup default CTest arguments when running from IDE
+
+   set(CMAKE_CTEST_ARGUMENTS "--output-on-failure;--verbose;${CMAKE_CTEST_ARGUMENTS}")
+
+   enable_testing()
+
+   #[[
+      add_unit_test(NAME name SOURCES file1 ... LIBRARIES lib1 ...)
+
+      Creates an executable called ${name}-test from the sources files ${file1}, ... and linked
+      to libraries ${lib1}, ... Catch2 is linked implicitly.
+
+      Create a CTest test called ${name}, that expects 0 code on success
+   ]]
+   function( add_unit_test )
+      cmake_parse_arguments(
+         ADD_UNIT_TEST # Prefix
+         "" # Options
+         "NAME" # One value keywords
+         "SOURCES;LIBRARIES"
+         ${ARGN}
+      )
+
+      if( NOT ADD_UNIT_TEST_NAME )
+         message( FATAL_ERROR "Missing required NAME parameter for the add_unit_test")
+      endif()
+
+      set( test_executable_name "${ADD_UNIT_TEST_NAME}-test" )
+
+      # Create test executable
+
+      add_executable( ${test_executable_name} ${ADD_UNIT_TEST_SOURCES} "${CMAKE_SOURCE_DIR}/tests/Catch2Main.cpp")
+      target_link_libraries( ${test_executable_name} ${ADD_UNIT_TEST_LIBRARIES} Catch2::Catch2 )
```

Comment:
Catch2 needs to link something?  I thought it is just one amalgamated header file.  Or is that precisely what this line does, just adding in include path.

Reply:
"Modern" CMake uses `target_link_libraries` for header-only libraries as well.

Reply:
Got it